### PR TITLE
cody-gateway: treat incoming traces as links instead of parents

### PR DESCRIPTION
**Pull request by bobheadxi**
Lately I've noticed many (most?) of our Sentry events come with `TraceId` that 404's on Cloud Trace, which kind of defeats the whole purpose of our tracing. I've spot-checked and see that all our traces in Cloud Trace have corresponding `Request` logs, but when attempting the inverse we often get a 404 in Cloud Trace.

I think the problem is that we are treating incoming traces as parents instead of linked traces. Treating the incoming request as a parent is cool because you get to see the whole thing end-to-end across projects, but it has the issue where half our traces are missing parents, and I think this is what causes logs to include traces that 404 when you query it on Cloud Trace

Was thinking of performing the parent/linking based on whether the request emits a ShouldTrace policy, but we have the same issue with on-prem customers etc if they enable tracing we'll end up with traces missing parents on our end that causes misbehaviour in Cloud Trace. I think I'm going to revert our usage of having remote traces be treated as parents, and just preserve them as links instead. This still gives us some bidirectional linking:

- In Sourcegraph/App/etc, we keep the return trace/span in trace attributes and also send it back to clients.
- In Cody Gateway, the span link will point us to the Sourcegraph trace

## Test plan

CI

### CODE REVIEW

File: enterprise/cmd/cody-gateway/shared/config.go
Comment (unknwon) [inline enterprise/cmd/cody-gateway/shared/config.go:1]: Are changes in this file belong to this PR? Looks like not related.

**Review by unknwon — APPROVED** [review]

File: enterprise/cmd/cody-gateway/shared/config.go
Comment (bobheadxi) [reply enterprise/cmd/cody-gateway/shared/config.go:1]: Ah yes, unrelated, will remove!

File: enterprise/cmd/cody-gateway/shared/config.go [hunk 1]
```diff
@@ -115,6 +118,8 @@ func (c *Config) Load() {
 
 	c.Trace.Policy = policy.TracePolicy(c.Get("CODY_GATEWAY_TRACE_POLICY", "all", "Trace policy, one of 'all', 'selective', 'none'."))
 	c.Trace.GCPProjectID = c.Get("CODY_GATEWAY_TRACE_GCP_PROJECT_ID", os.Getenv("GOOGLE_CLOUD_PROJECT"), "Google Cloud Traces project ID.")
+	c.Trace.QueueSize = c.GetInt("CODY_GATEWAY_TRACE_QUEUE_SIZE", "2048", "Maximum spans to queue for export.")
```
Comment (rafax) [inline enterprise/cmd/cody-gateway/shared/config.go:121]: I couldn't find how these two fields are used - if we have a problem exporting, will we keep up to 2048 spans, or will keep an unlimited number of spans in memory?

Comment (bobheadxi) [reply enterprise/cmd/cody-gateway/shared/config.go:121]: Sorry, these were unintentional additions, just removed them! for context I was thinking of changing some of the defaults, but there are some magic OTEL env vars you can use to change these instead, so if we still suspect traces being dropped I'll experiment with those instead
